l3kernel/l3regex.dtx

```diff
@@ -1135,10 +1135,10 @@
 %   Empty a \tn{toks} or set it to a value, given its number.
 %    \begin{macrocode}
 \cs_new_protected:Npn \@@_toks_clear:N #1
-  { \@@_toks_set:Nn #1 { } }
+  { \tex_toks:D #1 = { } }
 \cs_new_eq:NN \@@_toks_set:Nn \tex_toks:D
 \cs_new_protected:Npn \@@_toks_set:No #1
-  { \tex_toks:D #1 \exp_after:wN }
+  { \tex_toks:D #1 = \exp_after:wN }
```

Comment:
We should likely tidy up the `:D` use one way or the other!

Reply:
This is one of the core data structures (even if we don't use it that much in expl3) and as such I think it warrants a fast as possible implementation, so I'm for the `:D` functions including the use of `=` in the code. If you are worried I suggest to add a note at the start of the code section that the implementation is  coded for speed as potentially the functions are used very often.